test/translate_ligo.coffee

```diff
@@ -605,12 +604,90 @@ describe "translate ligo section", ()->
     type state is record
       reserved__empty_state : int;
     end;
-
+    
     function castType (const contractStorage : state) : (state) is
       block {
         const u : nat = abs(-(1));
         const i : int = int(abs(255));
         const addr : address = ("tz1ZZZZZZZZZZZZZZZZZZZZZZZZZZZZNkiRg" : address);
       } with (contractStorage);
     """#"
-    make_test text_i, text_o
\ No newline at end of file
+    make_test text_i, text_o
+  
+  it "this", ()->
+    text_i = """
+    pragma solidity ^0.5.0;
+    
+    contract This_test {
+      address owner;
+      
+      function _transferOwnership(address newOwner) public {
+        owner = newOwner;
+        this;
+      }
+      
+      function transferOwnership(address newOwner) public {
+        this._transferOwnership(newOwner);
+        owner = address(this);
+      }
+    }
+    
+    """#"
+    text_o = """
+    type state is record
+      owner : address;
+      reserved__initialized : bool;
+    end;
+    
+    type fix_underscore__transferOwnership_args is record
+      newOwner : address;
+    end;
+    
+    type transferOwnership_args is record
+      newOwner : address;
+    end;
+    
+    function fix_underscore__transferOwnership (const opList : list(operation); const contractStorage : state; const newOwner : address) : (list(operation) * state) is
+      block {
+        contractStorage.owner := newOwner;
+      } with (opList, contractStorage);
+    
+    function transferOwnership (const opList : list(operation); const contractStorage : state; const newOwner : address) : (list(operation) * state) is
+      block {
+        const tmp_0 : (list(operation) * state) = fix_underscore__transferOwnership(opList, contractStorage, newOwner);
+        opList := tmp_0.0;
+        contractStorage := tmp_0.1;
+        contractStorage.owner := self_address;
+      } with (opList, contractStorage);
+    
+    type router_enum is
+      | Fix_underscore__transferOwnersh of fix_underscore__transferOwnership_args
```

Comment:
Because such a long prefix leads to name truncation which may lead to even worse name collisions

Reply:
Is there way not to lock ourselves to identifiers like `fix_ownership` in tests?
It would be a trouble if we are to change them